cmd/kube-apiserver/app/server.go

```diff
@@ -133,7 +133,10 @@ func Run(s *options.APIServer) error {
 	}
 
 	// Proxying to pods and services is IP-based... don't expect to be able to verify the hostname
-	proxyTLSClientConfig := &tls.Config{InsecureSkipVerify: true}
+	proxyTLSClientConfig, err := transport.TLSConfigFor(s.KubeletConfig.TransportConfig())
```

Comment:
Without this fix it is impossible to use kubectl exec, because apiserver doesn't send its kubelet certificate to the kubelet.